step-capstone/src/components/Trip.js

```diff
@@ -1,256 +1,231 @@
-                            departureDate: data.departureDate,
-                            arrivalDate: data.arrivalDate,
-                            description: data.description
-                        }])
-                    })
-                    break;
-                default: throw "Invalid input";
-            }
-        }
-    }
-
-    render() {
-        return (
-            <div className="trip">
-                <Grid id="map-component"><Map zoom={13} center={{ lat: 51.5, lng: 0.087 }} /></Grid>
-                <Grid container className="foreground" direction="row" justify="space-between">
-                    <Grid item id="finalized-component">
-                        <Finalized
-                            list={this.state.items.filter((item) => item.finalized)}
-                            onRemoveItem={this.handleRemoveItem}
-                            onEditItem={this.handleEditItem}
-                            onAddItem={this.handleAddItem}
-                        />
-                    </Grid>
-                    <Grid item id="unfinalized-component">
-                        <Unfinalized
-                            list={this.state.items.filter((item) => !item.finalized)}
-                            onRemoveItem={this.handleRemoveItem}
-                            onEditItem={this.handleEditItem}
-                            onAddItem={this.handleAddItem}
-                        />
-                    </Grid>
-                </Grid>
-                <Grid id="add-button-component">
-                    <AddItemButton onAddItem={this.handleAddItem} />
-                </Grid>
-            </div>
-        );
-    }
+const tripSetting = {
+  destinations : ["London, Milan"],
+  startDate : new Date("2020-07-24T02:00:00"),
+  endDate : new Date("2020-07-26T02:00:00"),
+  description: ""
 }
-
-/*
- Classes for testing UI purposes
-*/
-function Finalized(props) {
-    return (
-        <Grid>
-            {
-                props.list.map((item) => {
-                    return <TravelObject
-                        key={item.id}
-                        data={item}
-                        onRemoveItem={props.onRemoveItem}
-                        onEditItem={props.onEditItem}
-                        onAddItem={props.onAddItem}
-                    />
-                })
-            }
-        </Grid>
-    )
+export default class Trip extends React.Component {
+  constructor(props) {
+      super(props);
+      this.state = {
+          items: [],
+          tripSetting: {
+              destination : [],
+              startDate : new Date(),
+              endDate : new Date(),
+              description : ""
+          }
+      }
+      this.handleRemoveItem = this.handleRemoveItem.bind(this);
+      this.handleEditItem = this.handleEditItem.bind(this);
+      this.handleAddItem = this.handleAddItem.bind(this);
+      this.handleEditTripSetting = this.handleEditTripSetting.bind(this);
+  }
+  fetchData() {
```

Comment:
Nit: It's nice to have extra newlines between function definitions, maybe put those back and use elsewhere.